stablecoin-v3/src/virtual_liquidity_pools.rs

```diff
@@ -0,0 +1,57 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::MEDIAN_POOL_DELTA;
+
+use super::config;
+
+pub type Block = u64;
+
+#[elrond_wasm::module]
+pub trait VLPModule: config::ConfigModule {
+    fn update_pool_delta(&self) {
+        let current_block = self.blockchain().get_block_nonce();
+        let last_replenish_block = self.last_replenish_block().get();
+        if current_block > last_replenish_block {
+            let median_delta = BigUint::from(MEDIAN_POOL_DELTA);
+            let pool_delta_mapper = self.pool_delta();
+            let pool_delta = pool_delta_mapper.get();
+            let replenish_rate = self.pool_recovery_period().get();
+            let replenish_value = if pool_delta > median_delta {
+                (&pool_delta - &median_delta) / replenish_rate
+            } else if median_delta > pool_delta {
+                (&median_delta - &pool_delta) / replenish_rate
+            } else {
+                BigUint::zero()
+            };
```

Comment:
Shouldn't this also take into account the difference between `current_block` and `last_replenish_block`? For instance, I'd expect `replenish_value` to be proportional to number of blocks since last replenish.